lib/typed_ecto_schema/type_builder.ex

```diff
@@ -65,14 +65,26 @@ defmodule TypedEctoSchema.TypeBuilder do
   end
 
   defmacro __define_type__(types, schema_opts) do
-    if Keyword.get(schema_opts, :opaque, false) do
-      quote bind_quoted: [types: types] do
-        @opaque t() :: %__MODULE__{unquote_splicing(types)}
-      end
-    else
-      quote bind_quoted: [types: types] do
-        @type t() :: %__MODULE__{unquote_splicing(types)}
-      end
+    use_typecheck? = TypeCheck.Macros in __CALLER__.requires
```

Comment:
Does this also work when you don't directly have `use TypeCheck`? e.g., `use MyOwnMacro` that generates the `use TypeCheck` part?

Reply:
Yes, because in order to have `@type!` working, `TypeCheck.Macros have to be required`.

Reply:
In fact, this will work even if you just do `require TypeCheck`